### PR TITLE
Remove the relationships param

### PR DESCRIPTION
**Pull request by hxoliverrumbelow**
This addresses #64 - stop namespacing the filtering of relationships into a `relationships` query parameter and merge that functionality into the `filter` query parameter.

### CODE REVIEW

File: lib/postProcessing/filter.js [hunk 1]
```diff
@@ -62,14 +62,50 @@ filter._filterMatches = function(textToMatch, propertyText) {
 filter._filterKeepObject = function(someObject, filters) {
   for (var k in filters) {
```
Comment (pmcnr-hx) [inline lib/postProcessing/filter.js]: Can we rename `k` to `attributeName` or something similar that makes clear what it is?

Comment (hxoliverrumbelow) [inline lib/postProcessing/filter.js]: Absolutely :+1:

**Comment by DuncanFenning** [comment]
Looks good :shipit:

File: lib/postProcessing/filter.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 "use strict";
 var filter = module.exports = { };
 
+var _ = require("underscore");
```
Comment (pmcnr-hx) [inline lib/postProcessing/filter.js]: Worth switching to `lodash`?

File: lib/postProcessing/filter.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 var debug = require("../debugging.js");
 
 filter.action = function(request, response, callback) {
-  var allFilters = request.params.filter;
+  var allFilters = _.extend({ }, request.params.filter);
```
Comment (pmcnr-hx) [inline lib/postProcessing/filter.js]: `_.clone(request.params.filter)` should achieve the same result for this case and read better. What do you think?

File: lib/postProcessing/filter.js [hunk 4]
```diff
@@ -60,16 +61,52 @@ filter._filterMatches = function(textToMatch, propertyText) {
...
+      if (!relationshipMatches) return false;
     }
-    if (!matchOR) return false;
   }
   return true;
 };
+
+filter._attributesMatchesOR = function(attributeValue, whitelist) {
+  var matchOR = false;
+  for (var j = 0; j < whitelist.length; j++) {
```
Comment (pmcnr-hx) [inline lib/postProcessing/filter.js]: `whitelist.forEach` instead and avoid the need for the index `j`?

File: lib/postProcessing/filter.js (hunk 2 above)
Comment (hxoliverrumbelow) [inline lib/postProcessing/filter.js]: Im up for it, but the move would be better done in its own pull request - we've got several other uses of underscore dotted around at the moment.

**Comment by pmcnr-hx** [comment]
This is great stuff. Just some very minor suggestions, that aren't blockers. :+1: :shipit:

File: lib/postProcessing/filter.js (hunk 3 above)
Comment (hxoliverrumbelow) [inline lib/postProcessing/filter.js]: I'll change this when we move to `lodash`, lodash.clone === underscore.extend

File: lib/postProcessing/filter.js (hunk 4 above)
Comment (hxoliverrumbelow) [inline lib/postProcessing/filter.js]: :+1:
